analyses/tcga-capture-kit-investigation/scripts/intersect-bed-maf.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+##############################
+# 
+# This script intersects all the BED and then use that to intersect with 
+# released PBTA and TCGA MAF and then counted all the mutation number within 
+# that intersection region, and then mapped that counts to the project and 
+# tumor type for each sample.
+# 
+##############################
+
+
+## set data path variable 
+DATA='../../data'
+SCRATCH='../../scratch'
+
+
+## get intersection between all the BED files
+bedtools intersect -a $DATA/WGS.hg38.strelka2.unpadded.bed -b $DATA/WGS.hg38.mutect2.vardict.unpadded.bed \
+| bedtools intersect -a - -b $DATA/WGS.hg38.lancet.unpadded.bed \
+| bedtools intersect -a - -b results/hg38lft-whole_exome_agilent_1.1_refseq_plus_3_boosters.targetIntervals.bed \
+| bedtools intersect -a - -b results/hg38lft-whole_exome_agilent_designed_120.targetIntervals.bed \
+| bedtools intersect -a - -b results/hg38lft-whole_exome_agilent_plus_tcga_6k.targetIntervals.bed \
+| bedtools intersect -a - -b results/hg38lft-tcga_6k_genes.targetIntervals.bed \
+> $SCRATCH/intersect-all-tcga_pbta.bed
```

Comment:
are we consider this to be in the final results? was reading the overall readme, looks like we have defined
> Files that are primarily tabular results files should be placed in a results subdirectory. Intermediate files that are useful within the processing steps but that do not represent final results should be placed in ../../scratch/.

I guess we probably should for the intersection in the results, but I always hesitated to put datafile rather than code in git, partially also bc the current OpenPBTA repo size is over 2.5G, and that makes me a little nervous. 

was looking at the top 10 unused file from the object idx from `.git/objects/pack/`, most of them are rds from previous analysis and we are not using it now?
```bash
$ git verify-pack -v .git/objects/pack/pack-4100e6ced849e5b0e2f9ae2c91da2ba5b9dae115.idx | sort -k3 -nr | head
870ed90f180072c89f7d9551f61d650be0a6f9c5 blob   85868950 13434311 550188084
ab29581d30cd0df08d9f031430a6896b46a9b93b blob   70759530 15725245 515885298
4028bd1f3680118d7d1e174394b977a48abd2baf blob   69873669 69894467 189615262
a72647329760dcf37d2a9b01f0fb77490dcf4401 blob   65817511 65804473 79933719
c0964ceaa319514f6aad78e4d0accd0ff291a17f blob   54589644 54584920 4608881
3bf9348bc05947ba145192a6dcf19801172048d3 blob   40942761 39855277 431034850
e1e8a5d128f619dfe57ed081099c52760f59f356 blob   40671573 15561340 531610543
b7cde15b9aca0746ab5caa0a7cf499bb5672b8f1 blob   40671561 15561331 636751664
e0cc36389dafa0b7a1f54e1d0979d48d4a4cfdd3 blob   39233596 24391072 153991957
b634015720525968f7f67c28b8be963581291ece blob   32662601 27167111 678012702

### grep those id and got
63132e59375cd82f824e92568c4321fccf4aac07 analyses/focal-cn-file-preparation/results/cnvkit_annotated_cn_autosomes.tsv.bz2
b80d9abc2c500751f1a30defe982134687ac5e08 analyses/focal-cn-file-preparation/results/controlfreec_annotated_cn_autosomes.tsv.bz2
af045f5bbe71db4a407be4975e3b0c9a6662c58a analyses/molecular-subtyping-HGG/hgg-subset/hgg_zscored_expression.stranded.RDS
a72647329760dcf37d2a9b01f0fb77490dcf4401 analyses/mutect2-vs-strelka2/results/combined_results.tsv.zip
335cb437d3a11b4a38f33290acba94df7347a450 analyses/molecular-subtyping-HGG/hgg-subset/hgg_zscored_expression.RDS
d5c117261828abe0da9df434abc02b881e47327d analyses/focal-cn-file-preparation/results/cnvkit_annotated_cn_autosomes.tsv.gz
91f6131381f06e6112b7475deb8568a0e693de04 analyses/focal-cn-file-preparation/results/cnvkit_annotated_cn_autosomes.tsv.gz
4028bd1f3680118d7d1e174394b977a48abd2baf analyses/focal-cn-file-preparation/results/cnvkit_annotated_cn_autosomes.tsv.gz
ab29581d30cd0df08d9f031430a6896b46a9b93b analyses/snv-callers/results/consensus/mutation_id_list.rds
870ed90f180072c89f7d9551f61d650be0a6f9c5 analyses/focal-cn-file-preparation/results/annotated_cn.tsv
```

Reply:
Committing files that are under the 100MB GitHub limit is not an uncommon practice in a data analysis project. Although there are downsides from a git performance perspective, the upside is that it would allow, as @jashapiro suggests, reviewers to spot check your files for anything that is unexpected. In addition, running everything that is required to produce an intermediate file locally in a reasonable amount of time may be unfeasible for a reviewer and we could get different files based on a difference in environment (depends on if you are using Docker). If you prefer to and are able to provide intermediate files in another way (e.g., S3), you can feel free to do that if you document how reviewers and other folks that come to the repository can get them.

Reply:
OK, cool. I think for now, I will then just keep the intersection BED in the `scratch/` for now as the bedtools is fast and eventually put the file in the data release, so ppl can grab it from S3. does that sound OK?

Reply:
I was suggesting that you provide the files to reviewers via an alternative mechanism like S3 at the time of review. What you suggest does not cover a scenario where someone wants to do development using these files prior to a data release.